cmd/git/main.go

```diff
@@ -93,7 +93,8 @@ func init() {
 	// Optional flag to be able to override the default shallow clone depth,
 	// which should be fine for almost all use cases we use the Git source step
 	// for (in the context of Shipwright build).
-	pflag.UintVar(&flagValues.depth, "depth", 1, "Create a shallow clone based on the given depth")
+	// Setting depth to 0 means no shallow clone (full clone)
+	pflag.UintVar(&flagValues.depth, "depth", 0, "Create a shallow clone with the given depth. 0 means full clone (no shallow).")
```

Comment:
This is a breaking change - current Shipwright users will see a performance hit if we switch from "no history" when cloning to "full history." I recommend reverting this change.